wandb/sdk/interface/artifacts/artifact_manifest.py

```diff
@@ -24,7 +25,10 @@ def __post_init__(self) -> None:
         self.path = util.to_forward_slash_path(self.path)
         self.extra = self.extra or {}
         if self.local_path and self.size is None:
-            raise ValueError("size required when local_path specified")
+            try:
+                self.size = Path(self.local_path).stat().st_size
+            except OSError as e:
+                raise ValueError("size required when local_path specified") from e
```

Comment:
Nit: I would remove the `try... catch...`.
- The message is directed at wandb developers, not users.
- Passing `size` won't help anyway, since you'll get the same error, just in a different place.

Reply:
Good point, the stack trace won't help the end user, and the FileNotFound error will be enough for us.